app/routes/judging.ts

```diff
@@ -14,88 +15,114 @@ judgingRouter.use(authenticate);
-    //     judgingToCriteria = new QualifierJudgingToCriteria();
-    // }
-
-    // judging.judgeId = ctx.state.user.id;
-    // judging.roundId = round.id;
-    // judging.submissionId = submission.id;
-    // judging.comment = comment;
-    // await judging.save();
-
-    // judgingToCriteria.criteria = criteria;
-    // judgingToCriteria.judging = judging;
-    // judgingToCriteria.score = score;
-    // await judgingToCriteria.save();
-
-    // return ctx.body = {
-    //     judging,
-    //     success: 'Saved',
-    // };
+    const round: Round = ctx.request.body.round;
+
+    if (!round) {
+        return ctx.body = { error: 'fuck' };
+    }
+
+    if (round.isQualifier) {
+        const submissionId = convertToIntOrThrow(ctx.request.body.submissionId);
+        const criteriaId = await convertToIntOrThrow(ctx.request.body.criteriaId);
+        const score = convertToIntOrThrow(ctx.request.body.score);
+        const comment = ctx.request.body.comment && ctx.request.body.comment.trim();
+
+        const [currentRound, criteria, submission] = await Promise.all([
+            Round.findCurrentJudgingRound(),
+            Criteria.findOneOrFail({ id: criteriaId }),
+            Submission.findOneOrFail({ id: submissionId }),
+        ]);
+
+        if (!score || !comment || !criteria || !submission) {
+            return ctx.body = { error: 'Missing data' };
+        }
+
+        if (!currentRound) {
+            return ctx.body = { error: 'There is currently no round to judge' };
+        }
+
+        if (score > criteria.maxScore) {
+            return ctx.body = { error: 'Score is higher than expected' };
+        }
+
+        let judging = await QualifierJudging.findOne({
+            judgeId: ctx.state.user.id,
+            submissionId: submission.id,
+        });
+
+        let judgingToCriteria;
+
+        if (judging) {
+            judgingToCriteria = await QualifierJudgingToCriteria.findOne({
+                criteria,
+                qualifierJudgingId: judging.id,
+            });
+        } else {
+            judging = new QualifierJudging();
+        }
+
+        if (!judgingToCriteria) {
+            judgingToCriteria = new QualifierJudgingToCriteria();
+        }
+
+        judging.judgeId = ctx.state.user.id;
+        judging.submissionId = submission.id;
+        await judging.save();
+
+        judgingToCriteria.criteria = criteria;
+        judgingToCriteria.qualifierJudgingId = judging.id;
+        judgingToCriteria.score = score;
+        judgingToCriteria.comment = comment;
+        await judgingToCriteria.save();
+
+        const judgingDone = await QualifierJudging.find({
+            where: { judgeId: ctx.state.user.id },
+            relations: ['qualifierJudgingToCriterias'],
+        });
+
+        return ctx.body = {
+            judgingDone,
+            success: 'Saved!',
+        };
+
+    } else {
+        // do something!!
+    }
+
+    ctx.body = { success: 'Saved!' };
 });
```

Comment:
remove this response ?,
also check if submission is part of current round somewhere